services/src/pro/datasets/postgres.rs

```diff
@@ -721,36 +765,46 @@ where
         let mut conn = self.conn_pool.get().await?;
         let tx = conn.build_transaction().start().await?;
 
-        self.ensure_permission_in_tx(dataset_id.into(), Permission::Owner, &tx)
-            .await
-            .boxed_context(crate::error::PermissionDb)?;
-
-        let stmt = tx
-            .prepare(
-                "
-        SELECT 
-            TRUE
-        FROM 
-            user_permitted_datasets p JOIN datasets d 
-                ON (p.dataset_id = d.id)
-        WHERE 
-            d.id = $1 AND p.user_id = $2 AND p.permission = 'Owner';",
-            )
-            .await?;
-
-        let rows = tx
-            .query(&stmt, &[&dataset_id, &self.session.user.id])
-            .await?;
-
-        if rows.is_empty() {
-            return Err(Error::OperationRequiresOwnerPermission);
+        let _uploaded = self.uploaded_dataset_status_in_tx(&dataset_id, &tx).await;
+        if let Err(error) = _uploaded {
+            if matches!(error, UnknownDatasetId) {
+                self.ensure_permission_in_tx(dataset_id.into(), Permission::Owner, &tx)
+                    .await
+                    .boxed_context(crate::error::PermissionDb)?;
+
+                let stmt = tx
+                    .prepare(
+                        "
+                SELECT
+                    TRUE
+                FROM
+                    user_permitted_datasets p JOIN datasets d
+                        ON (p.dataset_id = d.id)
+                WHERE
+                    d.id = $1 AND p.user_id = $2 AND p.permission = 'Owner';",
+                    )
+                    .await?;
+
+                let rows = tx
+                    .query(&stmt, &[&dataset_id, &self.session.user.id])
+                    .await?;
+
+                if rows.is_empty() {
+                    return Err(Error::OperationRequiresOwnerPermission);
+                }
+
+                let stmt = tx.prepare("DELETE FROM datasets WHERE id = $1;").await?;
+
+                tx.execute(&stmt, &[&dataset_id]).await?;
+
+                tx.commit().await?;
+
+                return Ok(());
```

Comment:
This is a bit odd without the option to delete data, but keep metadata. If the dataset is not a user upload, the engine would do that, otherwise the default now would but a full deletion.

Reply:
what exactly is werid about it?

Reply:
The previous way to delete data was to delete meta data, but keep data. That still exists for non-user uploads. But user uploads do not have that option.